android-components/components/support/utils/src/test/java/mozilla/components/support/utils/URLStringUtilsTest.kt

```diff
@@ -88,6 +88,40 @@ class URLStringUtilsTest {
         assertFalse(isURLLike("www.c-c-  "))
         assertFalse(isURLLike("3-3 "))
 
+        // IPv6 literals
+        assertTrue(isURLLike("[::]"))
+        assertTrue(isURLLike("[::1]"))
+        assertTrue(isURLLike("[1::]"))
+        assertTrue(isURLLike("[::]/"))
+        assertTrue(isURLLike("[1:2:3:4:5:6:7:8]"))
+        assertTrue(isURLLike("https:[::]"))
+        assertTrue(isURLLike("https://[::]"))
+        assertTrue(isURLLike("https://[::1]"))
+        assertTrue(isURLLike("https://[1::]"))
+        assertTrue(isURLLike("https://[2001:db8::1.2.3.4]/"))
+        assertTrue(isURLLike("https://[1:2:3:4:5:6:7:8]/"))
+        assertTrue(isURLLike("[2001:db8::1.2.3.4]/"))
+
+        // Invalid IPv6 literals, just to illustrate the limits of the regex.
+        assertFalse(isURLLike("::1"))
+        assertFalse(isURLLike(":::"))
+        assertFalse(isURLLike("[[http://]]"))
+        assertFalse(isURLLike("[[["))
+        assertFalse(isURLLike("[[[:"))
+        assertFalse(isURLLike("[[[:/"))
+        assertFalse(isURLLike("http://]]]"))
+        assertTrue(isURLLike("[:::"))
+        assertTrue(isURLLike("fe80::"))
+        assertTrue(isURLLike("x:["))
+        assertTrue(isURLLike("http://[::"))
+        assertTrue(isURLLike("http://[::/path"))
+        assertTrue(isURLLike("http://[::?query"))
+        assertTrue(isURLLike("[[http://banana]]"))
+        assertTrue(isURLLike("http://[[["))
+        assertTrue(isURLLike("[[[::"))
+        assertTrue(isURLLike("[[[::/"))
+        assertTrue(isURLLike("http://[1.2.3]"))
```

Comment:
I think this block includes tests that assert that `isURLLike` works as expected (returning false for malformed IPv6 literals), as well as tests showing the limits of the current regex (returning the wrong result for wellformed and malformed IPv6 literals). 

It would be a bit clearer to separate the two types of tests in two groups described with a comment.

Reply:
Also, we noticed that we could add tests with an URL with the wrong number of bytes : 
```kotlin 
        assertFalse(isURLLike("https://[1:2:3:4:5:6:7]/"))
        assertFalse(isURLLike("https://[1:2:3:4:5:6:7:8:9]/"))
```

Those test are currently failing. Would it be possible to adapt the regex so that we take this case into account? 

If not, we can put those tests in the block of tests showing the limits of the current regex.

Reply:
This regex makes no attempt to parse IP addresses, so cases like `[1:2:3:4:5:6:7]` belong in the "Invalid IPv6 literals incorrectly recognized as valid" bucket.